modules/heat_conduction/src/base/HeatConductionApp.C

```diff
@@ -56,6 +56,19 @@ associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
   registerSyntaxTask("ThermalContactAction", "ThermalContact/*", "add_material");
   registerSyntaxTask("ThermalContactAction", "ThermalContact/*", "add_secondary_flux_vector");
 
+  registerSyntaxTask(
+      "ThermalContactAction", "Modules/HeatConduction/ThermalContact/*", "add_aux_kernel");
+  registerSyntaxTask(
+      "ThermalContactAction", "Modules/HeatConduction/ThermalContact/*", "add_aux_variable");
+  registerSyntaxTask("ThermalContactAction", "Modules/HeatConduction/ThermalContact/*", "add_bc");
+  registerSyntaxTask(
+      "ThermalContactAction", "Modules/HeatConduction/ThermalContact/*", "add_dirac_kernel");
+  registerSyntaxTask(
+      "ThermalContactAction", "Modules/HeatConduction/ThermalContact/*", "add_material");
+  registerSyntaxTask("ThermalContactAction",
+                     "Modules/HeatConduction/ThermalContact/*",
+                     "add_secondary_flux_vector");
+
   registerSyntaxTask("RadiationTransferAction", "GrayDiffuseRadiation/*", "append_mesh_generator");
```

Comment:
This action should be under `/Modules/HeatConduction` as well... But that's for another PR. @snschune